src/packages/core/collection/collection.context.ts

```diff
@@ -30,107 +29,119 @@ export class UmbCollectionContext<ItemType, FilterModelType extends UmbCollectio
 	#filter = new UmbObjectState<FilterModelType | object>({});
 	public readonly filter = this.#filter.asObservable();
 
-	repository?: UmbCollectionRepository;
+	#views = new UmbArrayState<ManifestCollectionView>([]);
+	public readonly views = this.#views.asObservable();
 
-	/*
-	TODO:
-	private _search = new StringState('');
-	public readonly search = this._search.asObservable();
-	*/
+	#currentView = new UmbObjectState<ManifestCollectionView | undefined>(undefined);
+	public readonly currentView = this.#currentView.asObservable();
+
+	repository?: UmbCollectionRepository;
+	collectionRootPathname: string;
 
 	constructor(host: UmbControllerHostElement, entityType: string, repositoryAlias: string) {
-		this._entityType = entityType;
-		this._host = host;
-
-		new UmbObserverController(
-			this._host,
-			umbExtensionsRegistry.getByTypeAndAlias('repository', repositoryAlias),
-			async (repositoryManifest) => {
-				if (repositoryManifest) {
-					const result = await createExtensionApi(repositoryManifest, [this._host]);
-					this.repository = result as UmbCollectionRepository;
-					this._onRepositoryReady();
-				}
-			}
-		);
+		this.entityType = entityType;
+		this.host = host;
+
+		const currentUrl = new URL(window.location.href);
+		this.collectionRootPathname = currentUrl.pathname.substring(0, currentUrl.pathname.lastIndexOf('/'));
+
+		this.init = Promise.all([
+			new UmbObserverController(
```

Comment:
I know we still dont have the Context Base, but, ideally a Context is a Controller.
So make the CollectionContext extends the Controller base like this: `extends UmbBaseController`

This will give you the ability to do`this.observe()`

But even more importantly, it will make the Context a ControllerHost, meaning it owns the references to its sub controllers. meaning cleanup/destroying will be much more sharp. Enabling taking the CollectionContext out of service, without destroying the origin element. That might not make any difference in this case, but I want us to aim for that architecture :-)

Reply:
Great! This is now updated